src/mem/metaslab.h

```diff
@@ -1,20 +1,408 @@
 #pragma once
 
-#include "../backend/metatypes.h"
+#include "../backend/backend_concept.h"
 #include "../ds/helpers.h"
 #include "../ds/seqset.h"
-#include "../mem/remoteallocator.h"
 #include "freelist.h"
 #include "sizeclasstable.h"
 
 namespace snmalloc
 {
-  // The Metaslab represent the status of a single slab.
-  class alignas(CACHELINE_SIZE) Metaslab
+  struct RemoteAllocator;
+
+  /**
+   * Remotes need to be aligned enough that the bottom bits have enough room for
+   * all the size classes, both large and small. An additional bit is required
+   * to separate backend uses.
+   */
+  static constexpr size_t REMOTE_MIN_ALIGN =
+    bits::max<size_t>(CACHELINE_SIZE, SIZECLASS_REP_SIZE) << 1;
+
+  /**
+   * Base class for the templated MetaEntry.  This exists to avoid needing a
+   * template parameter to access constants that are independent of the
+   * template parameter.
```

Comment:
It now also includes the fields themselves, too (sort of like the short window from #477 up through #493, the very important difference being that the pagemap now contains the derived class -- or, one assumes, a descendant -- rather than the parent, so no concerns about UB and downcasting to the wrong type arise.  Yes?)

Reply:
I've updated the comment.